### ISSUE TITLE
Build failure: haskellPackages.hercules-ci-cnix-store

### ISSUE THREAD

**Issue by solomon-b** [NixOS/nixpkgs#294673]
### Steps To Reproduce

1. `nix-repl> :b pkgs.haskellPackages.hercules-ci-cnix-store`

### Build log

```
error: builder for '/nix/store/fbv2yjms21832jzzia2alylm2qgmnwxy-hercules-ci-cnix-store-0.3.4.0.drv' failed with exit code 1;
       last 10 log lines:
       > Dependency inline-c: using inline-c-0.9.1.10
       > Dependency inline-c-cpp: using inline-c-cpp-0.5.0.2
       > Dependency protolude: using protolude-0.3.3
       > Dependency temporary: using temporary-1.3
       > Dependency text: using text-2.0.2
       > CallStack (from HasCallStack):
       >   withMetadata, called at libraries/Cabal/Cabal/src/Distribution/Simple/Utils.hs:370:14 in Cabal-3.8.1.0:Distribution.Simple.Utils
       > Error: Setup: The pkg-config package 'nix-store' version >=2.4 && <2.17 is
       > required but the version installed on the system is version 2.18.1
       >
```

### Additional context

This occurs on 23.11.

### Notify maintainers

@roberth

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
➜ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9, NixOS, 23.11 (Tapir), 23.11.20240308.2be119a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/y0c95bwyvs80pm69hdd4b11pyq2ghiwh-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by roberth** [comment]
Fixed in #294579

The 23.11 channels haven't progressed to include it yet: https://nixpk.gs/pr-tracker.html?pr=294579

**Comment by solomon-b** [comment]
Thanks!
